heal-remedy.lic

```diff
@@ -0,0 +1,124 @@
+  Documentation: https://elanthipedia.play.net/Lich_script_repository#heal-remedy
+=end
+
+custom_require.call(%w[common common-healing common-items])
+
+
+arg_definitions = [
+  [
+    { name: 'debug', regex: /debug/i, optional: true }
+  ]
+]
+
+args = parse_args(arg_definitions)
+$debug_mode_hr = args.debug
+
+$remedies = get_data('remedies')
+
+def wounds
+  wounds = DRCH.check_health
+  pause 1
+  echo (wounds)  if $debug_mode_hr
+  hurt = wounds.select{|level, _| level >= 4}.values.flatten
+  wound_grouping = { "right arm" => "limbs", "left arm" => "limbs", "left leg" => "limbs", "right leg" => "limbs", "right hand" => "limbs", "left hand" => "limbs", "right eye" => "eyes", "left eye" => "eyes" }
+  scar_grouping = { "right arm" => "limbs", "left arm" => "limbs", "left leg" => "limbs", "right leg" => "limbs", "right hand" => "limbs", "left hand" => "limbs", "head" => "face", "neck" => "face", "eyes" => "face", "chest" => "body", "abdomen" => "body", "back" => "body" }
+  hurt = hurt.map{ |raw| wound_grouping[raw] || raw}.uniq
+  scar = hurt.map{ |raw| scar_grouping[raw] || raw}.uniq
+  echo ("Wounds: ") if $debug_mode_hr
+  if $debug_mode_hr
+    hurt.each do |key| 
+      puts key
+    end
+  end
+  echo()
+  echo ("Scars: ") if $debug_mode_hr  
+  if $debug_mode_hr
+    scar.each do |key| 
+      puts key
+    end
+  end
+  hurt.each do |key|
+    puts key if $debug_mode_hr
+    echo("In hurt loop") if $debug_mode_hr
+    apply_wounds(key)
+  end
+  pause 180 # pause 3 minutes before using scar remedies
+  scar.each do |key|
+    puts key if $debug_mode_hr
+    echo("In scar loop") if $debug_mode_hr
+    apply_scars(key)
+  end
+end
+  
+def apply_wounds(key)
+  echo("In apply_wounds function and key is #{key}") if $debug_mode_hr
+  rem_wounds = $remedies['remedies']['external'][key]
+  echo(rem_wounds) if $debug_mode_hr
+  rem_wounds.each do |key|
+    loop_debug
+    if DRCI.exists?(key) 
+      DRC.bput("get #{key}", "You get")
+      case key
+        when /salve/, /ungent/, /poultices/, /ointment/
+          DRC.bput("rub #{key}", "You")
+        when /potion/, /tonic/, /elixir/, /draught/
+          DRC.bput("drink #{key}", "You")
+       end
+       DRC.bput("stow #{key}", "You")
+    else
+      echo("*** No more #{key}! ***")  
+    end
+  end
+end
+
+def apply_scars(key)
+  echo("In apply_scar function and key is #{key}") if $debug_mode_hr
+  rem_scars = $remedies['remedies']['scars'][key]
+  echo(rem_scars) if $debug_mode_hr
+  rem_scars.each do |key|
+    loop_debug
+    if DRCI.exists?(key) 
+      DRC.bput("get #{key}", "You get")
+      case key
+        when /salve/, /ungent/, /poultices/, /ointment/
+          DRC.bput ("rub #{key}", "You")
+        when /potion/, /tonic/, /elixir/, /draught/
+          DRC.bput ("drink #{key}", "You")
+       end
+       DRC.bput("stow #{key}", "You")
+    else
+      echo("*** No more #{key}! ***")
+      DRC.bput("Trying general scar remedies.")
+      apply_general_scar(general)
```

Comment:
Where is `general` defined?

Reply:
the next def is where it is, line 99.

Reply:
That's where the function `apply_general_scar` is declared. I'm asking where the argument to that function, `general`, is declared.

Reply:
I want to pass the word general to the function so it goes to the general key in the yaml. Does it need quotes?

Reply:
Ah, I see. Yes, you'll need quotes around it. That will be a string with the contents 'general'. As written right now it's a variable named general (which doesn't exist).